src/main/java/net/fabricmc/loom/api/RunConfiguration.java

```diff
@@ -0,0 +1,169 @@
+
+	/**
+	 * Whether to append the project path to the {@link #getConfigurationName()} when {@code project} isn't the root project.
+	 *
+	 * <p>Warning: could produce ambiguous run config names if disabled, unless used carefully in conjunction with
+	 * {@link #getConfigurationName()}.
+	 */
+	Property<Boolean> getAppendProjectPathToConfigName();
+
+	/**
+	 * The main class of the run configuration.
+	 */
+	Property<String> getMainClass();
+
+	/**
+	 * The source set to use for the run configuration.
+	 */
+	Property<SourceSet> getSourceSet();
+
+	/**
+	 * The run directory for this configuration, relative to the root project directory.
+	 */
+	DirectoryProperty getRunDirectory();
+
+	/**
+	 * When true a run configuration file will be generated for IDE's.
+	 *
+	 * <p>By default only run configs on the root project will be generated.
+	 */
+	Property<Boolean> getGenerateRunConfig();
+
+	/**
+	 * Group this run config under the given folder.
+	 *
+	 * <p>This is currently only supported on IntelliJ IDEA.
+	 *
+	 * @return The property used to set the config folder.
+	 */
+	Property<String> getIdeConfigFolder();
+
+	/**
+	 * The true entrypoint, this is usually dev launch injector.
+	 * This should not be changed unless you know what you are doing.
+	 */
+	@ApiStatus.Experimental
+	Property<String> getDevLaunchMainClass();
+
+	default void inherit(RunConfiguration parent) {
+		getConfigurationName().convention(parent.getConfigurationName());
+		getJVMArguments().convention(parent.getJVMArguments());
+		getProgramArguments().convention(parent.getProgramArguments());
+		getEnvironmentVars().convention(parent.getEnvironmentVars());
+		getRuntimeEnvironment().convention(parent.getRuntimeEnvironment());
+		getAppendProjectPathToConfigName().convention(getAppendProjectPathToConfigName());
+		getMainClass().convention(parent.getMainClass());
+		getSourceSet().convention(parent.getSourceSet());
+		getRunDirectory().convention(parent.getRunDirectory());
+		getGenerateRunConfig().convention(parent.getGenerateRunConfig());
+		getIdeConfigFolder().convention(parent.getIdeConfigFolder());
+	}
+
+	/**
+	 * Configure run config with the default client options.
+	 */
+	default void client() {
+		getRuntimeEnvironment().convention("client");
+		getMainClass().convention(Constants.Knot.KNOT_CLIENT);
+
+		if (Platform.CURRENT.isRaspberryPi()) {
+			getEnvironmentVars().put("MESA_GL_VERSION_OVERRIDE", "4.3");
+		}
+	}
+
+	/**
+	 * Configure run config with the default server options.
+	 */
+	default void server() {
+		getProgramArguments().add("nogui");
+		getRuntimeEnvironment().convention("server");
+		getMainClass().convention(Constants.Knot.KNOT_SERVER);
+	}
+
+	default void property(String name, String value) {
+		getJVMArguments().add("-D%s=%s".formatted(name, value));
+	}
+
+	default void property(String name) {
+		getJVMArguments().add("-D%s".formatted(name));
+	}
+
+	default void properties(Map<String, String> props) {
+		props.forEach(this::property);
+	}
```

Comment:
A map for the properties is likely better, then they can be passed via the DLI config.